core/deployment/src/main/java/io/quarkus/deployment/recording/RecordingAnnotationsUtil.java

```diff
@@ -33,30 +32,32 @@ final class RecordingAnnotationsUtil {
             }
         }
 
-        IGNORED_PROPERTY_ANNOTATIONS = List.copyOf(ignoredPropertyAnnotations);
-        RECORDABLE_CONSTRUCTOR_ANNOTATIONS = List.copyOf(recordableConstructorAnnotations);
+        IGNORED_PROPERTY_ANNOTATIONS = ignoredPropertyAnnotations.toArray(new Class[0]);
+        RECORDABLE_CONSTRUCTOR_ANNOTATIONS = recordableConstructorAnnotations.toArray(new Class[0]);
```

Comment:
As we're making things more efficient, we can make `new Class[0]` a constant, no?

Reply:
I'd say it's unnecessary - compiler knows this pattern well. Also, any static constant takes memory forever :P

Reply:
>  compiler knows this pattern well

kk